### PR TITLE
[1.7.x] Added support for batch producing and handling batch of messages

### PR DESCRIPTION
**Pull request by vsvp21**

### CODE REVIEW

**Comment by mateusjunges** [comment]
I'm wondering why the tests pipeline is not running for this PR

**Comment by vsvp21** [comment]
> I'm wondering why the tests pipeline is not running for this PR

I have fixed it, forgot to add extra line at the end of configuration

File: src/BatchRepositories/BatchRepositoryInterface.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Junges\Kafka\BatchRepositories;
+
+use \RdKafka\Message;
+use Illuminate\Support\Collection;
+
+interface BatchRepositoryInterface
```
Comment (mateusjunges) [inline src/BatchRepositories/BatchRepositoryInterface.php:10]: This class should be in `src/Contracts`

File: src/BatchRepositories/BatchRepositoryInterface.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
```
Comment (mateusjunges) [inline src/BatchRepositories/BatchRepositoryInterface.php:3]: Please remove strict types from all classes

File: src/Config/BatchConfigInterface.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Junges\Kafka\Config;
```
Comment (mateusjunges) [inline src/Config/BatchConfigInterface.php:5]: Move this to `src/Contracts`

File: src/Producers/Producer.php [hunk 4]
```diff
@@ -53,6 +54,42 @@ public function produce(KafkaProducerMessage $message): bool
...
+    /**
+     * @throws CouldNotPublishMessage
+     */
+    public function produceBatch(MessageBatch $messageBatch): int
+    {
+        $topic = $this->producer->newTopic($this->topic);
+
+        $messagesIterator = $messageBatch->getMessages();
+
+        $messagesIterator->setIteratorMode(\SplDoublyLinkedList::IT_MODE_FIFO);
```
Comment (mateusjunges) [inline src/Producers/Producer.php:73]: Please add to the `use` statements

**Review by mateusjunges — CHANGES_REQUESTED** [review]

**Comment by mateusjunges** [comment]
@rtuin can you let me know your thoughts about this?

File: src/Producers/Producer.php (hunk 4 above)
Comment (vsvp21) [reply src/Producers/Producer.php]: @mateusjunges fixed

File: src/Config/BatchConfigInterface.php (hunk 3 above)
Comment (vsvp21) [reply src/Config/BatchConfigInterface.php]: @mateusjunges fixed

File: src/BatchRepositories/BatchRepositoryInterface.php (hunk 2 above)
Comment (vsvp21) [reply src/BatchRepositories/BatchRepositoryInterface.php]: @mateusjunges fixed

File: src/BatchRepositories/BatchRepositoryInterface.php (hunk 1 above)
Comment (vsvp21) [reply src/BatchRepositories/BatchRepositoryInterface.php]: @mateusjunges fixed

**Comment by vsvp21** [comment]
@mateusjunges should I fix something else? Or it is fine for now?

**Comment by mateusjunges** [comment]
@vsvp21 one more thing:  can you add a simple PHP doc block at least to the interface methods, explaining what each one is responsible for?

**Comment by vsvp21** [comment]
> @vsvp21 i think its fine now. I've just removed some final classes. For my open source code, i assume users know they are responsible for writing tests for any overwritten behaviour. I'll review it one more time later today, and i'll let you know if there is any changes to be made.

Okay, Thanks

**Comment by vsvp21** [comment]
> @vsvp21 one more thing: can you add a simple PHP doc block at least to the interface methods, explaining what each one is responsible for?

Yeah I'm gonna add it later today

**Comment by vsvp21** [comment]
@mateusjunges I have added PHP doc blocks

**Comment by mateusjunges** [comment]
I'll merge this later today or tomorrow, i need to update my website to generate the docs from branch `v1.7.x` before. Thank you @vsvp21.

**Review by mateusjunges — APPROVED** [review]

**Pull request merged**